### PR TITLE
[filebeat][azure-blob-storage] - Cleaned up documentation errors & fixed a minor bug.

### PR DESCRIPTION
**Pull request by ShourieG**
## Type of change
- Bug
- Docs

## Proposed commit message
There was a minor logical error and a documentation error for the PR: https://github.com/elastic/beats/pull/36690,
which was discovered after the merge. This PR aims to fix those.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Relates #36690

### CODE REVIEW

File: x-pack/filebeat/input/azureblobstorage/job.go [hunk 1]
```diff
@@ -175,6 +175,7 @@ func (j *job) readJsonAndPublish(ctx context.Context, r io.Reader, id string) er
 			if err := j.splitEventList(j.src.ExpandEventListFromField, item, offset, j.hash, id); err != nil {
 				return err
 			}
+			continue
```
Comment (bhapas) [inline x-pack/filebeat/input/azureblobstorage/job.go:178]: Anyway to hold this in a test? Looks like if we remove this it still goes through

Comment (ShourieG) [reply x-pack/filebeat/input/azureblobstorage/job.go:178]: @bhapas the nature of this test is such that right now it's not possible to do a negative fail test with the current test suit. I will look into how we can improve the test suit to incorporate such negative tests in future.

**Comment by ShourieG** [comment]
@bhapas if all else looks good, could you approve ?

File: x-pack/filebeat/input/azureblobstorage/job.go (hunk 1 above)
Comment (bhapas) [reply x-pack/filebeat/input/azureblobstorage/job.go:178]: Can we test this manually , if automation suite does not support now?

Comment (ShourieG) [reply x-pack/filebeat/input/azureblobstorage/job.go:178]: yes manually tested this on my local already and is behaving as expected. This is basically a copy of the same logic present in the s3 input here: https://github.com/elastic/beats/blob/a55ae5857c0d80e0834c9f3e98fd1b833675f12c/x-pack/filebeat/input/awss3/s3_objects.go#L220

Comment (bhapas) [reply x-pack/filebeat/input/azureblobstorage/job.go:178]: Great...!

**Review by bhapas — APPROVED** [review]
LGTM.. 

@ShourieG tested the change manually

**Pull request merged**